app/views/components/_topic_list.html.erb

```diff
@@ -15,13 +20,30 @@
 %>
 <% if items.any? %>
   <%= tag.ul(class: ul_classes, lang: "en") do %>
-    <% items.each do |item| %>
+    <% items.each_with_index do |item, index| %>
       <li class="app-c-topic-list__item">
         <%=
           link_to(
             item[:text],
             item[:path],
-            data: item[:data_attributes],
+            data: {
+              **item[:data_attributes] || {},
+              ga4_link: {
+                event_name: "navigation",
+                type: "document list",
+                index: {
+                  index_link: index + 1,
+                  # If ga4_data[:section_index] and ga4_data[:section_count] have not been passed to this component, we don't want to 
+                  # include null values in the push to the dataLayer. Therefore, we conditionally include these keys.
+                  **(ga4_data[:section_index] ? { index_section: ga4_data[:section_index] } : {}),
+                  **(ga4_data[:section_count] ? { index_section_count: ga4_data[:section_count] } : {}),
```

Comment:
This logic is somewhat complex for the view. I'm wondering if there's an easier approach?

```Ruby
# topic_list_helper.rb
ga4_data = {}
ga4_data[:index_section_count] = list_count if list_count
ga4_data[:index_section] = list_index + 1 if list_index
ga4_data[:section] = list_title if list_title
```

```Ruby
# _topic_list.html.erb
<li class="app-c-topic-list__item">
  <% 
     index_attributes = ga4_data
     index_attributes[:index_link] = index + 1
  %>
  <%= link_to(
    # etc.
    data: {
       ga4_link: {
            event_name: "navigation",
            index: index_attributes,
```

You'd have to figure something out for `section` as that isn't inside `index`, but it should be doable.

Reply:
This was along the lines of my initial approach but I felt it was a bit clunky. I'll revert it to how it was before.

Reply:
Great, thanks. I think it's better to try to keep the logic out of the view as much as possible.